sympy/parsing/latex/lark/transformer.py

```diff
@@ -107,37 +113,74 @@ def group_curly_parentheses(self, tokens):
         return tokens[1]
 
     def eq(self, tokens):
-        return sympy.Eq(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Eq(tokens[0], tokens[2])
+        else:
+            return sympy.Eq(tokens[0], tokens[2], evaluate=False)
 
     def ne(self, tokens):
-        return sympy.Ne(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Ne(tokens[0], tokens[2])
+        else:
+            return sympy.Ne(tokens[0], tokens[2], evaluate=False)
 
     def lt(self, tokens):
-        return sympy.Lt(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Lt(tokens[0], tokens[2])
+        else:
+            return sympy.Lt(tokens[0], tokens[2], evaluate=False)
 
     def lte(self, tokens):
-        return sympy.Le(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Le(tokens[0], tokens[2])
+        else:
+            return sympy.Le(tokens[0], tokens[2], evaluate=False)
 
     def gt(self, tokens):
-        return sympy.Gt(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Gt(tokens[0], tokens[2])
+        else:
+            return sympy.Gt(tokens[0], tokens[2], evaluate=False)
 
     def gte(self, tokens):
-        return sympy.Ge(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Ge(tokens[0], tokens[2])
+        else:
+            return sympy.Ge(tokens[0], tokens[2], evaluate=False)
 
     def add(self, tokens):
-        return sympy.Add(tokens[0], tokens[2])
+        if TransformToSymPyExpr.evaluate:
+            return sympy.Add(tokens[0], tokens[2])
+        else:
+            return sympy.Add(tokens[0], tokens[2], evaluate=False)
 
     def sub(self, tokens):
         if len(tokens) == 2:
+            # we want to parse things like -3.14 without an unevaluated
+            # multiplication showing up in the parse tree. This is very
+            # important when parsing things like "\sin^{-1} x" because
+            # using a `evaluate=False` multiplication node will make
+            # it show up as:
+            # `sympy.sin(x, sympy.Mul(-1, 1, evaluate=False), evaluate=False)`,
+            # which is not useful to anyone.
             return -tokens[1]
```

Comment:
I'm not really sure about you should do this
`srepr(parse_latex_lark('-(-1)', evaluate=False))` gives `Integer(1)` instead of `-(-1)` unevaluated.

Reply:
I may suggest to selectively evaluate only for `Integer` or `Rational` with positive value